File: mooc_application/src/main/webapp/js/service/roleMenuRService.js
//服务层
app.service('roleMenuRService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../roleMenuR/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../roleMenuR/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../roleMenuR/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../roleMenuR/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../roleMenuR/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../roleMenuR/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../roleMenuR/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: mooc_application/src/main/java/com/hongxuan/mooc/service/RoleMenuRService.java
package com.hongxuan.mooc.service;
import com.hongxuan.pojo.TbRoleMenuR;
import entity.PageResult;

import java.util.List;
/**
 * 业务逻辑接口
 * @author LiaoHongXuan
 *
 */
public interface RoleMenuRService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbRoleMenuR> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbRoleMenuR roleMenuR);
	
	
	/**
	 * 修改
	 */
	public void update(TbRoleMenuR roleMenuR);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbRoleMenuR findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbRoleMenuR roleMenuR, int pageNum,int pageSize);
	
}
